flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ArchivedExecutionVertex.java

```diff
@@ -100,13 +99,12 @@ public TaskManagerLocation getCurrentAssignedResourceLocation() {
         return currentExecution.getAssignedResourceLocation();
     }
 
-    @Nullable
     @Override
-    public ArchivedExecution getPriorExecutionAttempt(int attemptNumber) {
-        if (attemptNumber >= 0 && attemptNumber < priorExecutions.size()) {
-            return priorExecutions.get(attemptNumber);
-        } else {
-            throw new IllegalArgumentException("attempt does not exist");
-        }
+    public ExecutionHistory getExecutionHistory() {
+        return executionHistory;
+    }
+
+    static ExecutionHistory getCopyOfExecutionHistory(ExecutionVertex executionVertex) {
```

Comment:
can be private

Reply:
It will be used by `ArchivedSpeculativeExecutionVertex` in a latter pr(#20082), I would keep its default scope to avoid back&forth changes.